.github/workflows/publish-image-branch.yml

```diff
@@ -0,0 +1,103 @@
+name: 'Publish image to GHCR on push to branch'
+
+on:
+  push:
+    branches-ignore:
+      - 'release'
+
+env:
+  TARGET_IMAGE: "dbp-keycloak"
+
+jobs:
+  build_and_upload_image:
+    name: Publish image
+    runs-on: ubuntu-latest
+    outputs:
+      digest: ${{ steps.docker_build_push.outputs.digest }}
+    permissions:
+      packages: write
+      contents: read
+    strategy:
+      matrix:
+        variant: ['generic', 'erwin']
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+      
+      - name: Build image name and tags
+        id: docker_meta_img
+        uses: docker/metadata-action@v5
+        with:
+          images: | 
+            name=ghcr.io/${{ github.repository_owner }}/${{ env.TARGET_IMAGE }}-${{ matrix.variant }}
```

Comment:
Only an suggestion not an requierd change.
Use `${{ github.repository }}` instad of `${{ github.repository_owner }}/${{ env.TARGET_IMAGE }}` it will go to the same as `dBildungsplattform/dbp-keycloak` but you don't need an env var for the taget repo in the name and change if the gitrepo change.